src/plugins/initiatives/createGraph.js

```diff
@@ -90,27 +113,58 @@ export function createWeightedGraph(
     }
 
     // Generic approach to adding edges when the reference detector has a hit.
-    const edgeHandler = (
-      edges: $ReadOnlyArray<URL> | EdgeSpec,
+    const urlHandler = (
+      urls: $ReadOnlyArray<URL>,
       createEdge: EdgeFactoryT
     ) => {
-      // TODO: this is a temporary implementation, which takes an EdgeSpec and
-      // just takes the $ReadOnlyArray<URL>. This is only done to allow support
-      // for graphing NodeEntries to be added in a separate commit.
-      const urls = Array.isArray(edges) ? edges : edges.urls;
-
       for (const url of urls) {
         const addr = refs.addressFromUrl(url);
         if (!addr) continue;
         graph.addEdge(createEdge(initiative, addr));
       }
     };
 
+    // Generic approach to handling EdgeSpecs.
+    const edgeSpecHandler = (
+      {urls, entries}: EdgeSpec,
+      createEdge: EdgeFactoryT,
+      field: NodeEntryField
+    ) => {
+      // Delegate handling the URLs.
+      urlHandler(urls, createEdge);
+
+      for (const entry of entries) {
+        // Add the NodeEntry contribution itself to the graph.
+        const node = nodeFromEntry(entry, initiative.id, field);
```

Comment:
nit: would be easier to understand if this were called `entryNode` rather than just `node`, since there are a few different nodes in play here

Reply:
Updated :+1: